### ISSUE TITLE
Photon: srcset in 4.0.4 is being too smart and messing up my layout

### ISSUE THREAD

**Issue by rothschild86** [Automattic/jetpack#4175]
Here's the layout before 4.0.4:

![2016-06-20_171704](https://cloud.githubusercontent.com/assets/4555882/16210685/fc7b4e86-3702-11e6-97b4-e7f238825a08.png)

Here's after:

![2016-06-20_171618](https://cloud.githubusercontent.com/assets/4555882/16210693/06cdf226-3703-11e6-9207-243dae10a310.png)

Photon decided to introduce a max-width of 1000...

Disabled Photon for now, but would love to get it back.

You can see my specific code here - http://rothschildportraiture.com - built with Beaver Builder.

#### What I expected
Smooth minor update

#### What happened instead
Full width designs got messed up

#### Steps to reproduce the issue
Enable photon. Use beaver builder to add a full width image. (adding a row with a full width image background still works though...)

**Comment by kraftbj** [comment]
@rothschild86 - Can you try adding this to your theme's child theme or a functionality plugin:

```
add_filter( 'jetpack_content_width', 'set_jp_cw' );
function set_jp_cw( $width ){
   if ( ! $width ){
      $width = 2000;
   }
   return $width;
}
```

We put `1000` as a default for themes that have not set a `$content_width` for use with WordPress' responsive images.

**Comment by rothschild86** [comment]
Yeah, that worked fine. Thank you for a response, Brandon. I'm sure I won't be the last person to ask this.

**Comment by kraftbj** [comment]
Reopening for further investigation and examination of themes that do not set a content width.

One thought: If there's not a content width, do nothing. This would roll back the enhancements added in 4.0.4 (which attempted to prevent, in some cases, some extremely large images from being downloaded by the browser).

**Comment by rothschild86** [comment]
For the record, theme used: Genesis: Altitude Pro.

Is there a way to set it to device width instead of hard coding a value?

**Comment by kraftbj** [comment]
I'm working with StudioPress (I'm in their private repo) since they're the biggest shop that I've seen not actively use `content_width`. 

You could set it to 99999999. For the purposes of Photon, it would max out at the images full-size. I haven't, to be frank, tested that approach on other pieces of Jetpack that user the Jetpack filter for the content width, though. Might be a place we need to add a new filter to target this usage specifically.

**Comment by proudfoot** [comment]
I can confirm that setting `jetpack_content_width` to 99999999 causes `jetpack-video-wrapper` to set a height of 0

**Comment by kraftbj** [comment]
We're still discussing it, but may look at only implementing a default width when srcset'ing strictly for images within the post_content, not others., which may reach the goal of providing better performance for all sites while avoiding issues like this.

**Comment by kraftbj** [comment]
The sizes array: We should likely abort out for images not within `the_content`.

**Comment by kraftbj** [comment]
@rothschild86 - FYI - I haven't tested 4226 against your setup (Beaver Builder does appear to have a free version for testing), but I'm not expecting it to resolve it since those images are still within the content. 4226 fixes a related, but not wholly duplicative issue

**Comment by rothschild86** [comment]
BB Lite is free - https://wordpress.org/plugins/beaver-builder-lite-version/

Thinking of it... It would be nice if skipping could be activated by setting a certain css class on an image (very easy to do with BB)...

meanwhile, a mental note to self - tweak the original hack above to activate only on certain pages that use BB and take advantage of srcset on the rest of the site...

**Comment by rothschild86** [comment]
so for now the hack looks like this:

```
add_filter( 'jetpack_content_width', 'set_jp_cw' );
function set_jp_cw( $width ){
  $full_width_page_ids = array(2147,2170);
  
   if ( ! $width && in_array(get_the_ID(), $full_width_page_ids)){
      $width = 2000;
   }
   return $width;
}
```

**Comment by proudfoot** [comment]
I'm using a theme that was customized to allow specific images to be oversized within `the_content`, using a version of the `outdentImages` function in Intergalactic ( [lines 36-61 in intergalactic.js](https://themes.trac.wordpress.org/browser/intergalactic/1.3.2/js/intergalactic.js?rev=62790&order=name#L37) )  The theme separately declares `$content_width` and `$jetpack_content_width` so the added class is specific to those images.

Is there any way to filter the default independent of `$content_width` _and_ `$jetpack_content_width`?

**Comment by jeherve** [comment]
Also reported here:
https://wordpress.org/support/topic/422-photon-resizing-to-wrong-size/#post-8209438

**Comment by jeherve** [comment]
Also reported here:
https://wordpress.org/support/topic/image-resize-crop-and-rotation-problems-with-photon/

**Comment by jeherve** [comment]
Also reported here:
https://wordpress.org/support/topic/image-width-in-single-post/

**Comment by ntpixels** [comment]
Reported in 3309777-t

**Comment by jeherve** [comment]
Also in 1057305-zen

**Comment by htdat** [comment]
1615590-zen

**Comment by jeherve** [comment]
Also in pcLjiI-9e-p2
